### DISCUSSION TITLE
Import JSON as Part of a Metamodel

### DISCUSSION THREAD

**Discussion by danyeaw** [textX/textX#413] [Q&A]
Thanks for the awesome project :heart:!

I am trying to load a JSON file that has a metamodel defined and import all the namespace that it has in to another metamodel. I was able to follow the [docs example](https://textx.github.io/textX/3.1/multimetamodel/#use-case-referencing-non-textx-meta-modelsmodels-with-a-json-file) to create a metamodel from a JSON file and then access it via a model using the access keyword.

The example is doing the following:
JSON -> my_metamodel -> my_model

I want to do something like:
JSON -> json_metamodel -> my_metamodel

Is something like this possible? I have been struggling to combine the multi-metamodel concept with the JSON example.

**Comment by igordejanovic** [comment]
Hi @danyeaw. Sorry for this *very* late reply. Somehow this issue slipped through the cracks.

If I understand your issue, you would like to load a meta-model that is in a JSON format into a textX meta-model? 

That is not possible. You can only use scoping/reference resolving to resolve references inside your textX model to refer to data in non-textX sources, like the example from the docs demonstrates. The only way to describe a meta-model is by textX grammar language.

If textX could do something like that it would need to overlay its language concepts over a JSON format resulting in an ugly meta-language which would be ok for persistence by not very pleasant for manual writing or reading.

If my understanding of what you are trying to achieve is correct, I wonder why would you need something like that?

Moving this issue to discussions.
